file_catalog/server.py

```diff
@@ -651,8 +656,17 @@ async def put(self, uuid: str) -> None:
         # Validate Incoming Metadata
         if self.validation.has_forbidden_attributes_modification(self, metadata, db_file):
             return
-        if await pathfinder.contains_existing_filepaths(self, metadata, uuid=uuid):
+
+        # Deconflict with DB Records
+        # NOTE - PUT should not conflict with any existing record (excl. uuid's record)
+        # NOTE - by existing location(s) or by existing file-version
+        if await deconfliction.any_location_in_db(self, metadata.get("locations"), skip=uuid):
             return
+        try:
+            if await deconfliction.FileVersion(metadata).is_in_db(self, skip=uuid):
+                return
+        except deconfliction.IndeterminateFileVersionError:
+            pass
```

Comment:
Do we want to pass here?

I understand passing in the PATCH case, because we may not provide checksum and/or logical_name in a PATCH body, depending on what we're patching.

The PUT route is 'replace the resource' and a valid replacement shouldn't be throwing us an IndeterminateFileVersionError due to one of those fields being missing?

Reply:
I agree with @blinkdog

Reply:
Great catch. I've re-shuffled the validation & deconfliction logic in a new commit. I also replaced `pass` with an error for good measure.